src/go/util/token_generator.go

```diff
@@ -31,19 +31,28 @@ var (
 	tokenMux   = sync.Mutex{}
 )
 
-func GenerateAccessTokenFromFile(serviceAccountKey string) (string, time.Duration, error) {
+func GenerateAccessTokenFromFile(saFilePath string) (string, time.Duration, error) {
 	if token, duration := activeAccessToken(); token != "" {
 		return token, duration, nil
 	}
 
-	data, err := ioutil.ReadFile(serviceAccountKey)
+	data, err := ioutil.ReadFile(saFilePath)
 	if err != nil {
 		return "", 0, err
 	}
 
 	return generateAccessToken(data)
 }
 
+// A test-friendly version of `GenerateAccessTokenFromFile`
+func generateAccessTokenFromData(saData []byte) (string, time.Duration, error) {
+	if token, duration := activeAccessToken(); token != "" {
+		return token, duration, nil
+	}
+
```

Comment:
No it was not working. Reading from the file is ok, but writing to it does not work. I just disabled the test in Google3 manually. But with the new config versioning implementation, disabling tests is too complex.

Reply:
Agreed that adding code for that looks bad. Unfortunately, I'm not sure of any other workaround (other than to refactor the production code and unit test individual components, not the whole thing).

Reply:
I see. In this case, wrap `generateAccessTokenFromFile` by `GenerateAccessTokenFromFile` with comments, instead of copying code.

Only testing `GenerateAccessTokenFromFile` is fine.

Reply:
That will change production code, it will require reading the SA file every time you check the cache. The only option is to split the unit test into one for the cache and one for the generation. What do you think?

Reply:
SG